llvm/lib/Target/ARM/MCTargetDesc/ARMAsmBackend.cpp

```diff
@@ -587,6 +587,14 @@ unsigned ARMAsmBackend::adjustFixupValue(const MCAssembler &Asm,
         return 0;
     return 0xffffff & ((Value - 8) >> 2);
   case ARM::fixup_t2_uncondbranch: {
+    if (STI->getTargetTriple().isOSBinFormatCOFF() && !IsResolved &&
+        Value != 4) {
+      // MSVC link.exe and lld do not support this relocation type
+      // with a non-zero offset. ("Value" is offset by 4 at this point.)
+      Ctx.reportError(Fixup.getLoc(),
+                      "cannot perform a PC-relative fixup with a non-zero "
+                      "symbol offset");
+    }
     Value = Value - 4;
```

Comment:
I think it would be slightly cleaner to just move it one line bellow, where `Value` is adjusted by 4 and skip the offset explanation comment.

Reply:
Yes, I considered that, but as only 1 case out of 3 have the `Value = Value - 4;` as a separate step, I preferred to keep it consistent, and early upfront in each case.

Reply:
Sure, that's fine with me.